tests/foreman/sys/test_katello_certs_check.py

```diff
@@ -145,8 +201,75 @@ def test_positive_validate_katello_certs_check_output(self, cert_setup, cert_dat
             )
         self.validate_output(result, cert_data)
 
+    @pytest.mark.tier1
+    def test_katello_certs_check_output_wildcard_inputs(self, generate_certs, cert_data):
+        """Validate that katello-certs-check generates correct output with wildcard certs.
+
+        :id: 7f9da806-5b23-11eb-b7ea-d46d6dd3b5b2
+
+        :steps:
+
+            1. Get valid wildcard certs from generate_certs
+            2. Run katello-certs-check with the required valid arguments
+               katello-certs-check -c CERT_FILE -k KEY_FILE -r REQ_FILE
+               -b CA_BUNDLE_FILE
+            3. Assert the output has correct commands with options
+
+        :expectedresults: katello-certs-check should generate correct commands
+         with options.
+        """
+        with get_connection() as connection:
+            result = connection.run(
+                'katello-certs-check -c {} -k {} -b {}'.format(
+                    "certs/wildcard.crt",
+                    "certs/wildcard.key",
+                    "certs/ca.crt",
+                ),
+                output_format='plain',
+            )
+        self.validate_output(result, cert_data)
+
+    @pytest.mark.parametrize("error, cert_file, key_file, ca_file", invalid_inputs)
+    @pytest.mark.tier1
+    def test_katello_certs_check_output_invalid_input(
+        self, generate_certs, error, cert_file, key_file, ca_file
+    ):
+        """Validate that katello-certs-check raise the correct errors for invalid
+         inputs
+
+        :id: 37742f5e-598a-11eb-a349-d46d6dd3b5b2
+
+        :steps:
+
+            1. Get invalid certs from generate_certs
+            2. Run katello-certs-check with the required valid arguments
+               katello-certs-check -c CERT_FILE -k KEY_FILE -r REQ_FILE
+               -b CA_BUNDLE_FILE
+            3. Assert the output has correct error with message
+
+        :expectedresults: Katello-certs-check should raise error when it receives the invalid
+         inputs.
+        """
+        with get_connection() as connection:
+            result = connection.run(
+                'katello-certs-check -c {} -k {} -b {}'.format(
+                    cert_file,
+                    key_file,
+                    ca_file,
+                ),
+                output_format='plain',
+            )
+            results = result.stdout.split("\n\n")
+            for each_result in results:
+                if error['check'] in each_result:
+                    assert '[FAIL]' in each_result
+                    assert error['message'] in " ".join([result.stdout, result.stderr])
```

Comment:
here I'm using `result.stdout, result.stderr` not wanted to use the same variable

Reply:
Understood, I think that's even more reason to update the variable naming here to be more clear.

```
certs_check_result = connection.run('katello_certs_check <etc>')
for each result in certs_check_result.stdout.split('\n\n'):
    if error['check'] in result:
        assert '[FAIL]' in result
        assert error['message'] in f'{certs_check_result.stdout} {certs_check_result.stderr}'
```

Reply:
okay more cleaner, I like it